### PR TITLE
fix(types): add pool to ConnectionManager interface

### LINKED ISSUE #16756: Pool is missing from ConnectionManager interface
## Issue Creation Checklist

- [x] I understand that my issue will be automatically closed if I don't fill in the requested information
- [x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Bug Description

connectionManager is missing ```pool``` under ConnectionManager interface

image

[Documentation](https://sequelize.org/docs/v7/other-topics/connection-pool/) indicates that under connectionManager there should be pool available. It works correctly in my codebase and I'm able to use this value, however getting that Typescript error mentioned above. I suspect that this could type could be missing there.

image

### Reproducible Example

- Initialize db 
- Try accessing ```pool``` under connectionManager

image

### What do you expect to happen?

No Typescript error should show up

### What is actually happening?

I'm getting a Typescript error and have to use ```//@ts-ignore``` in order to ignore it

### Environment

- Sequelize version: 6.3.5
- Node.js version: 18.16.1
- If TypeScript related: TypeScript version: 5.2.2
- Database & Version: PostgreSQL 15.4
- Connector library & Version: pg 8.4.2

## Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [x] Yes, I have the time but I will need guidance.
- [ ] No, I don't have the time, but my company or I are [supporting Sequelize through donations on OpenCollective](https://opencollective.com/sequelize).
- [ ] No, I don't have the time, and I understand that I will need to wait until someone from the community or maintainers is interested in resolving my issue.

---

*Indicate your interest in the resolution of this issue by adding the 👍 reaction. Comments such as "+1" will be removed.*

### PR DESCRIPTION
**Pull request by madguy02**
## Pull Request Checklist

- [ ] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

closes #16756 , we updated the typings in v6 to support pool in Connection Manager interface

### CODE REVIEW

**Comment by madguy02** [comment]
@WikiRik please take a look at it, whenever possible

File: src/dialects/abstract/connection-manager.d.ts [hunk 1]
```diff
@@ -36,4 +36,10 @@ export interface ConnectionManager {
    * Destroys a pooled connection and removes it from the pool.
    */
   destroyConnection(conn: Connection): Promise<void>;
+
+  /**
+   * to be more strict, we can add type: {[key: string]: unknown}
+   * should we do it ?
+   */
+  pool: object;
```
Comment (WikiRik) [inline src/dialects/abstract/connection-manager.d.ts:44]: I'll need to take a look at the actual code but don't we define specific properties in the code of the dialects? Allowing any string as a key does not seem like much added value

Comment (madguy02) [reply src/dialects/abstract/connection-manager.d.ts:44]: yeah makes sense actually, removing the comment

Comment (madguy02) [reply src/dialects/abstract/connection-manager.d.ts:40]: "don't we define specific properties in the code of the dialects" - can you elaborate on this @WikiRik

**Comment by WikiRik** [comment]
@ephys mentioned that this does not seem to be a public API in v6 so I think we do not need this PR after all. Thanks for working on this though!

**Pull request closed without merging**